xarray/core/alignment.py

```diff
@@ -927,8 +927,8 @@ def _get_broadcast_dims_map_common_coords(args, exclude):
         for dim in arg.dims:
             if dim not in common_coords and dim not in exclude:
                 dims_map[dim] = arg.sizes[dim]
-                if dim in arg.coords:
-                    common_coords[dim] = arg.coords[dim].variable
+                if dim in arg.xindexes:
+                    common_coords.update(arg.xindexes[dim].create_variables())
```

Comment:
```suggestion
                if dim in arg._indexes:
                    common_coords.update(arg.xindexes.get_all_coords(dim))
```

The 1st suggestion is nit picking: no need to create an `Indexes` object for checking if there's an item, we can use the internal raw dict instead.

The 2nd suggestion is to use the `Indexes` API to directly get the relevant coordinate variables instead of re-creating them from the Xarray index.